ocf_datapipes/transform/xarray/pv/assign_daynight_status.py

```diff
@@ -0,0 +1,73 @@
+"""
+This is a class function that assigns day night status
+"""
+import logging
+
+import numpy as np
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+logger = logging.getLogger(__name__)
+
+uk_daynight_dict = {
+    1: [7, 17],
+    2: [7, 18],
+    3: [9, 19],
+    4: [6, 20],
+    5: [5, 21],
+    6: [4, 22],
+    7: [5, 22],
+    8: [5, 21],
+    9: [6, 20],
+    10: [7, 21],
+    11: [7, 17],
+    12: [7, 17],
+}
+logger.info(
+    f"The day and night standard hours are set by {'https://www.timeanddate.com/sun/uk/london'}"
+)
+
+
+@functional_datapipe("assign_daynight_status")
+class AssignDayNightStatusIterDataPipe(IterDataPipe):
+    """Adds a new dimension of day/night status"""
+
+    def __init__(self, source_datapipe: IterDataPipe):
+        """This method adds extra coordinate of day night status.
+
+        Args:
+            source_datapipe: Datapipe emiiting Xarray Dataset
+        Result:
+            <xarray.Dataset>
+            Dimensions:   (datetime: 289)
+            Coordinates:
+            * datetime  (datetime) datetime64[ns]
+            * status_day (datetime)
+        """
+
+        self.source_datapipe = source_datapipe
+
+    def __iter__(self) -> xr.DataArray():
+        """Returns an xarray dataset with extra dimesion"""
+
+        # Reading the Xarray dataset
+        for xr_dataset in self.source_datapipe:
+
+            # Getting Month and Hour values from time_utc and stacking them
+            date_month = np.asarray(xr_dataset.time_utc.dt.month.values, dtype=int)
+            date_hr = np.asarray(xr_dataset.time_utc.dt.hour.values, dtype=int)
```

Comment:
Values is already a numpy array so you shouldn't need this conversion.

Reply:
yes, changing right now!